flang/lib/Optimizer/Dialect/FIROps.cpp

```diff
@@ -1898,6 +1898,47 @@ fir::GlobalOp fir::createGlobalOp(mlir::Location loc, mlir::ModuleOp module,
   return result;
 }
 
+bool fir::valueHasFirAttribute(mlir::Value value,
```

Comment:
Is this any arbitrary value? Or should this only be on function arguments?

Reply:
It's intended to work on arbitrary values.
For instance, CONTIGUOUS can also be on a module pointer, so we will want to be able to check this on the result of the fir.address_of ops too, I think.
My main assumption described at the declaration spot in FIROpsSupport.h is that Fortran attributes will be reflected in mlir arguments, or fir.alloca/fir.allocmem of the entity, or fir.global/fir.address_of. So that what is where this helper is looking for the attribute.